src/main/java/io/codearte/jfairy/producer/person/PersonProvider.java

```diff
@@ -86,31 +100,66 @@ public Person get() {
 
 		//fixme - should be created only if needed
 		if (company == null) {
-			company = companyProvider.get();
+			company = companyFactory.produceCompany().get();
 		}
 
-		String firstName = dataMaster.getValuesOfType(FIRST_NAME, sex.name());
-		String middleName = randomBoolean() ? dataMaster.getValuesOfType(FIRST_NAME, sex.name()) : "";
-		String lastName = dataMaster.getValuesOfType(LAST_NAME, sex.name());
-		String email = generateEmail(firstName, lastName);
-		String username = generateUsername(firstName, lastName);
-		if (telephoneNumberFormat == null) {
-			telephoneNumberFormat = dataMaster.getRandomValue(TELEPHONE_NUMBER_FORMATS);
+		if (firstName == null) {
+			firstName = dataMaster.getValuesOfType(FIRST_NAME, sex.name());
+		}
+		if (middleName == null) {
+			middleName = randomBoolean() ? dataMaster.getValuesOfType(FIRST_NAME, sex.name()) : "";
+		}
+		if (lastName == null) {
+			lastName = dataMaster.getValuesOfType(LAST_NAME, sex.name());
+		}
+		if (email == null) {
+			email = generateEmail(firstName, lastName);
+		}
+		if (username == null) {
+			username = generateUsername(firstName, lastName);
+		}
+
+		if (telephoneNumber == null) {
+			if (telephoneNumberFormat == null) {
+				telephoneNumberFormat = dataMaster.getRandomValue(TELEPHONE_NUMBER_FORMATS);
+			}
+			telephoneNumber = baseProducer.numerify(telephoneNumberFormat);
+		}
+
+		if (dateOfBirth != null) {
+			age = Years.yearsBetween(dateOfBirth, DateTime.now()).getYears();
 		}
-		String telephoneNumber = baseProducer.numerify(telephoneNumberFormat);
 		if (age == null) {
 			age = baseProducer.randomBetween(MIN_AGE, MAX_AGE);
 		}
 		if (dateOfBirth == null) {
 			dateOfBirth = generateDateOfBirth();
 		}
-		String companyEmail = stripAccents(lowerCase(firstName + '.' + lastName + '@' + company.domain()));
-		// FIXME: Replace this with baseProducer
-		String password = RandomStringUtils.randomAlphanumeric(8);
 
-		return new Person(firstName, middleName, lastName, addressProvider.get(), email,
+		if (companyEmail == null) {
+			companyEmail = stripAccents(lowerCase(firstName + '.' + lastName + '@' + company.domain()));
+		}
+		if (password == null) {
+			// FIXME: Replace this with baseProducer
```

Comment:
I can do it though I think it would be better to fix this in another PR. It would be nice to have #68 merged or closed before further changes since it may cause lots of merge conflict otherwise.